### DISCUSSION TITLE
How to scale out worker node without k8s?

### DISCUSSION THREAD

**Discussion by AliscaCL** [milvus-io/milvus#15754] [Q&A and General discussion] [answered]
For some network safety reason, I have to deploy a milvus cluster in offline environment. 
Of course, there is no k8s cluster in offline environment. 
So what shall I do now? How to register the worker node in other server?
I have 5 or more server, such as:
192.168.0.100~105
I have raise milvuse cluster in 192.168.0.100 with docker-compose. What shall I do in next?

**Comment by xiaofan-luan** [comment]
you can just bring up querynode and they will register to etcd automatically and the balance will happen automatically.

However without etcd if querynode crash then it will not be brought up

**Reply by AliscaCL** [reply]
Thanks, but if I start a milvus cluster in Server A. When I bring up query node in server B, how the etcd register the worker node from server B to A?

**Reply by xiaofan-luan** [reply]
If you want to lookup into the details, it's in Session_Util.go.

If you want to learn more about how to use it, see https://milvus.io/docs/v2.0.x/scaleout.md

**Reply by AliscaCL** [reply]
Oh... yeah, I have already tried to follow the doc described in scaleout.md,  but I am not famaliar with the deploying a k8s cluster in an offline environment. So, I am trying to deploy milvus manually with docker compose in cluster mode. But I can't find the details about how to register a worker node manually. Which container should I config with? Proxy or coor node?

**Reply by xiaofan-luan** [reply]
to deploy a offline cluster, you will need to:
1. bring up etcd
2. bring up pulsar
3. bring up minio
4. bring up rootcoord
5. bring up datacoord
6. bring up querycoord
7. bring up. indexcoord
8. bringup datanode, indexnode,querynode
9. bringup proxy
which is a long journey.....

If you have fairly small amount of data, we recommend you to deploy standalone in a offline cluster which you only need to do yum install. So far we don't offer any convenient way to deploy milvus distributed cluster offline

**Comment by LoveEachDay** [comment, marked as answer]
If you want to add more nodes into your milvus cluster, you don't need modify any config, just run a new service with docker-compose. The new nodes will automatically register itself into the etcd (make sure your etcd service is accessible for your new node service), and everything is done.

There is no need to change config for proxy or coord.

**Reply by AliscaCL** [reply]
Wow, thanks for your answer at first. 
So, you means, if I have 5 host server, 192.168.0.100 ~ 192.168.0.105.
I should start a cluster mode milvus in 192.168.0.100 at first, and then docker-compose up milvus service in other 4 server.
Nothing should be done to join these 5 host server together?

**Reply by LoveEachDay** [reply]
You first start up the dependencies like etcd, pulsar and minio. 
Then modify the config for milvus providing the etcd, pulsar and minio endpoints.
Then start up the milvus cluster components using the same `milvus.yaml` .

**Reply by AliscaCL** [reply]
Thank you so much. 
I will try it asap~

Thanks again.

**Reply by waiyiaisg** [reply]
Could you detail the changes needed in the config for milvus etc?
